src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/SaveTempDataPropertyFilter.cs

```diff
@@ -1,14 +1,26 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
 using System.Reflection;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Internal;
 
 namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
 {
-    public class SaveTempDataPropertyFilter : ISaveTempDataCallback
+    public class SaveTempDataPropertyFilter : ISaveTempDataCallback, IActionFilter
     {
-        public string Prefix { get; set; }
+        private readonly ITempDataDictionaryFactory _factory;
+
+        internal IList<PropertyHelper> PropertyHelpers { get; set; }
+
+        public SaveTempDataPropertyFilter(ITempDataDictionaryFactory factory)
+        {
+            _factory = factory;
+        }
+
+        public string Prefix = "TempDataProperty-";
```

Comment:
@rynowak Wondering if we need this prefix...a scenario that I am thinking is for keep. If a value is bound to a tempdata property, it's as though its read and would be marked for deletion, but if a user chooses to keep it, then he might want to explicitly say `TempData.Keep(nameof(OrderStatusMessage))` (where OrderStatusMessage is a TempData property).